### PR TITLE
E2E Phase 2.3: networkidle/sleep sweep, lint rules to error

### PR DESCRIPTION
**Pull request by delano**
> [!NOTE]
> Originally stacked on #3412 (Phase 2.1+2.2), which **merged to `develop`** while this PR was being prepared. The merge preserved PR 3's commits, so this diff is already PR 4-only against `develop` — no rebase needed. Marked ready for review.

PR 4 of the [E2E remediation plan](https://github.com/onetimesecret/onetimesecret/blob/claude/e2e-phase24-networkidle-sweep/e2e/docs/e2e-remediation-plan.md) — Phase 2.3: replace every timing-guess wait with a deterministic signal, then make the lint ban blocking.

## What this does

Sweeps all **341** lint findings (`298 playwright/no-networkidle` + `43 playwright/no-wait-for-timeout`) in four per-directory commits, then flips both rules to `'error'`:

| Commit | Scope | Warnings cleared |
|--------|-------|------------------|
| `e2e/all` | 4 files | 29 |
| `e2e/auth` | 2 files | 16 |
| `e2e/full` | 19 files | 262 |
| `e2e/full-billing` | 4 files | 34 |
| `eslint.config.ts` | rules → `'error'` | — |

### Replacement patterns (chosen per call-site, not blanket)

- **`networkidle` after `goto()`/`reload()`** (253 sites, the mechanical class) → the app-readiness flag: `await expect(page.locator('html[data-app-ready="true"]')).toBeAttached()` (canonical usage `e2e/global.setup.ts`). The flag is set in `src/main.ts` after `consumeBootstrapData` + mount + brand theme + `router.isReady()` — strictly stronger than network quiescence.
- **`networkidle` after in-SPA clicks / `goBack()`/`goForward()`** → `waitForURL` / web-first `toHaveURL` on the known destination route. The readiness flag is per-document-load and would no-op on same-document router navigations, so URL assertions are the honest wait there.
- **Sleeps before reading route-handler capture flags** → `expect.poll(() => flag)` on the very flag the next line asserted; form saves anchor on the mocked round-trip via `waitForResponse`; signup submissions anchor on `waitForRequest('**/auth/create-account**')`.
- **UI-settle sleeps** → web-first assertions on the actual condition: toggle flips poll `isToggleEnabled` to the target state; billing-interval tab switches wait on `aria-pressed='true'` (verified against `src/apps/workspace/billing/PlanSelector.vue`); MFA recovery-mode switch waits on either outcome element via `locator.or()`.
- **`__BOOTSTRAP_ME__` readiness-polling** (Phase 2.2's "tests wait on that" half) → gone. The six redundant `waitForFunction(... === true)` polls in `brand-customization` are deleted (consumption happens before mount, so the flag implies it); `integration.spec.ts`'s bootstrap-consumption test still **asserts** the consumption marker (that contract is the test's point) but waits on the flag. *Content* reads/writes of `__BOOTSTRAP_ME__` (sso-csrf shrimp checks, secret-context branding injection) remain, per plan.
- A few checks got **stronger**, not just deterministic: `secret-context`'s console-error listeners were registered *after* navigation (sampling an arbitrary 1s window and missing load errors) — they're hoisted before `goto`; `scope-switcher` TC-SS-054's 1s sleep guarded no assertion at all and is deleted outright.

### Lint enforcement

`playwright/no-networkidle` and `playwright/no-wait-for-timeout` flip from `'warn'` to `'error'` (`eslint.config.ts:609-610`) — new timing-guess waits now fail lint instead of accruing silently. The block comment documents the replacement primitives.

### Tracker hand-off

`e2e/docs/e2e-remediation-plan.md` updated: status line, Progress + PR-sequence tables, Phase 2 item 3 marked swept, and the pickup section rewritten for PR 5 (Phase 2.4 defensive-skip triage) with tree-verified pointers — the 436-site `test.skip` census by class/directory, the default-workspace guarantee behind the 66 `!org` guards, and the 12 deliberately-kept credential skips.

## Quarantine

Nothing quarantined in this PR: `e2e/QUARANTINE.md` stays empty because no test has yet *demonstrated* brokenness under the new waits — the container suite can't run in the dev environment. If this PR's CI run surfaces genuinely-broken tests, they get `test.fixme` + a QUARANTINE.md row (owner + issue) per the Phase 1 process, not a re-skip; the flaky gate stays blocking.

## Verified locally

- `pnpm lint:e2e`: **0 problems** with both rules at `'error'` (baseline was 341 warnings).
- `pnpm lint:quick`: exit 0. `pnpm run type-check` (vue-tsc): exit 0.
- `playwright --list`: 415 tests in 30 files, all projects load.
- Per-directory `tsc --noEmit --strict` parity: identical pre-existing error sets before/after each sweep commit (2 in `all/`, 4 in `auth/`, 5 in `full/`, 4 in `full-billing/` — these specs aren't in the project tsconfig; no new errors introduced).
- `grep -rn 'networkidle\|waitForTimeout' e2e/ --include='*.spec.ts'` → no call sites (only the lint-rule comment and one doc reference remain).

## Deferred to CI

The container suite (built OCI image + Redis) can't run locally; the behavioral proof of the new waits is this PR's `container-e2e-tests` run, now triggering against `develop`. Expect the readiness-flag waits to be *less* flaky than networkidle, not more — but any genuine failures get quarantined per the process above.

https://claude.ai/code/session_01WRor5yk5SGGE7KWg1bFf99

---
_Generated by [Claude Code](https://claude.ai/code/session_01WRor5yk5SGGE7KWg1bFf99)_

### CODE REVIEW

File: e2e/full/scope-switcher.spec.ts [hunk 1]
```diff
@@ -541,19 +545,16 @@ test.describe('Scope Switcher - Switching Behavior', () => {
...
-          // Wait for update
-          await page.waitForTimeout(500);
-
-          // Verify trigger now shows new org
-          const newOrgName = await trigger.textContent();
-          expect(newOrgName).not.toBe(currentOrgName);
+          // Verify trigger now shows new org (web-first assertion waits
+          // for the switcher to update)
+          await expect(trigger).not.toHaveText(currentOrgName ?? '');
         }
```
[the remark this answers is not in the document]
Comment (delano) [reply e2e/full/scope-switcher.spec.ts]: Fixed in af3eb9c9: the name is now read as `(await trigger.textContent())?.trim() ?? ''` with an explicit `expect(currentOrgName).not.toBe('')` guard, and the post-switch wait uses `not.toContainText(currentOrgName)` as suggested — a null/empty read now fails the test loudly instead of degrading the assertion.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: e2e/full/scope-switcher.spec.ts [hunk 2]
```diff
@@ -872,10 +875,9 @@ test.describe('Scope Switcher - Edge Cases', () => {
...
         await menuItems.nth(1).click();
-        await page.waitForTimeout(1000);
 
-        // Domain scope may have been reset if new org doesn't have that domain
-        // This is expected behavior
+        // Domain scope may have been reset if new org doesn't have that
+        // domain - expected behavior; nothing further to assert here.
       }
     }
   });
```
[the remark this answers is not in the document]
Comment (delano) [reply e2e/full/scope-switcher.spec.ts:887]: Deliberately deferred rather than fixed here: TC-SS-054 was assertion-less before this sweep too (the 1s sleep guarded nothing), and the honest fix needs a product-behavior decision — whether switching orgs must reset the domain scope unconditionally or only when the new org lacks the domain. That "tests that cannot fail" triage is exactly the mandate of the Phase 2.4 skip-triage PR (in flight, stacked on this branch), which will either add the real assertion or `test.fixme` it with a QUARANTINE.md row per the remediation plan. Leaving this thread open as the tracking marker until that PR lands.

---
_Generated by [Claude Code](https://claude.ai/code)_
